src/main.rs

```diff
@@ -110,6 +111,7 @@ struct Args {
     flag_skip_readme: bool,
     flag_code: u8,
     flag_stdout: bool,
+    flag_reachable_link: bool,
```

Comment:
cc @drahnr 
silly question, where can I pick up the result of what is passed in the flag? Can I pick up in the markdown file?

Reply:
Can you elaborate where you need what information? I don't quite understand what you are asking for.

The `Args` struct contains all the commandline parameters. From there on you have to thread 🧵 / stitch 🪡 it through to the place you need it. If there are already other arguments passed through, consider adding a per component argument struct.

I hope that covers it :) if not, please elaborate on you question :question:

Reply:
I actually do not know how to or where should I save the boolean value of this flag from main to markdown file.
Should I do something like this? or how can I do: save the boolean value of the flag so I can do actions in markdown about whether the links should be checked for Get request or not?


```
lazy_static! {
    static ref USERS: HashMap<&'static str, Vec<&'static str>> = {
        let mut map = HashMap::new();
        map.insert("Loe", vec!["user", "user"]);
        map
    };
}

fn show_user(name: &str) {
    let access = USERS.get(name);
    println!("{}: {:?}", name, access);
}
```
Thank you in advance

Reply:
> markdown file

I assume you mean propagate the `bool` from `main.rs` / `fn main` to `markdown.rs`.

Just looked into the code now, probably should have done that earlier.

Soo, this is indeed a bit iffy. There are three options (see detailed discussion underneath):

1. fiddle the `bool` through to `erase_cmark`
2. make the whole configuration a global singleton
3. make the link destination check a separate, indepedent checker that implements `trait Checker`

I would recommend option `3.`, since markdown parsing is pretty cheap compared to the http lookup (even with `smid` yet to be enabled), and also compared to the dictionary lookup (see #57 for the flamegraph)

Option `1.` is feasible, but would mix concerns rather unpleasantly - a mashup of a feature and a parsing layer - not so good and entangles the code quite a bit.

Option `2.` is something for a server application / service mostly done for ad-hoc re-configuration, which would work too, but is somewhat special and makes testing harder, so I think the trade of is simply not given here.

Reply:
I did not really follow how adding the Checker trait will help me to have the desired flag in markdown file.

This is an example of the config I found in the Hunspell checker like here:
https://github.com/drahnr/cargo-spellcheck/blob/master/src/checker/hunspell.rs#L118

`{ lang: Some("en_US"), search_dirs: SearchDirs(Some(["/home/tmhdev/Documents/cargo-spellcheck/.config", "/usr/share/hunspell"])), extra_dictionaries: Some(["/home/tmhdev/Documents/cargo-spellcheck/.config/lingo.dic"]), quirks: Some(Quirks { transform_regex: Some([WrappedRegex(^'([^\s])'$), WrappedRegex(^[0-9]+x$), WrappedRegex(^\#[0-9]+$), WrappedRegex(^[0-9]+$), WrappedRegex(^.+\+$)]), allow_concatenation: Some(true), allow_dashes: None }) }`

Also, I am confused if you meant that this flag `reachable_link` should not be a boolean but rather be passed to:
`  --checkers=<checkers>     Calculate the intersection between`

I am confused about this step. Probably because I lack to understand how other parts of the project work.

Reply:
I did not get notifed of this comment :raised_eyebrow: maybe one of the plenty gh outages swallowed that notification, maybe I screwed up.

The issue here is, that currently the markdown is a layer in the general processing of input - which as of today - has no knowledge of any configuration, it's in it's core a pseudo-stream based pipeline with origin tracking.


Now if you would implement this with the `trait Checker` you would be called straight from main like `Hunspell` and `LanguageTool` (and soon `Reflow`) are, each with their defined params. So if you move away from being just a knot or layer in the input processing, but actually a consumer of the output of those layers, you can operate on the "raw" chunks (with markdown included) and do the checks right there.